lib/std/array_list.zig

```diff
@@ -240,23 +240,39 @@ pub fn ArrayListAligned(comptime T: type, comptime alignment: ?u29) type {
         pub fn replaceRange(self: *Self, start: usize, len: usize, new_items: []const T) Allocator.Error!void {
             const after_range = start + len;
             const range = self.items[start..after_range];
+            if (range.len < new_items.len) {
+                const first = new_items[0..range.len];
+                const rest = new_items[range.len..];
+                @memcpy(range[0..first.len], first);
+                try self.insertSlice(after_range, rest);
+            } else {
+                self.replaceRangeAssumeCapacity(start, len, new_items);
+            }
+        }
+
+        /// Replace range of elements `list[start..][0..len]` with `new_items`.
+        /// If `len < new_items.len` then it asserts that `.capacity` is
+        /// large enough for the increase in items.
+        /// Invalidates pointers if this ArrayList is resized.
+        pub fn replaceRangeAssumeCapacity(self: *Self, start: usize, len: usize, new_items: []const T) void {
+            const after_range = start + len;
+            const range = self.items[start..after_range];
 
             if (range.len == new_items.len)
                 @memcpy(range[0..new_items.len], new_items)
             else if (range.len < new_items.len) {
                 const first = new_items[0..range.len];
                 const rest = new_items[range.len..];
-
                 @memcpy(range[0..first.len], first);
-                try self.insertSlice(after_range, rest);
+                const dst = self.addManyAtAssumeCapacity(after_range, rest.len);
+                @memcpy(dst, rest);
             } else {
                 @memcpy(range[0..new_items.len], new_items);
                 const after_subrange = start + new_items.len;
 
                 for (self.items[after_range..], 0..) |item, i| {
                     self.items[after_subrange..][i] = item;
                 }
-
                 self.items.len -= len - new_items.len;
```

Comment:
This is missing a `@memset` to set things to `undefined`. Note that `orderedRemove` *was* doing this but is no longer since it is calling this function now.

Reply:
Done - thanks for catching that